docs: documentation update (#125)
kyma-project/kyma-companion: README.md

poetry remove {package_name}
```

## Create and Use Virtual Environments with Poetry

You can create a virtual environment for the project by navigating to the project's root directory and running the following command:

```bash
poetry install
```

This will create a new virtual environment and install the project's dependencies.

### Activate the Virtual Environment

> To activate the virtual environment, use `poetry shell`. This will start a new shell session with the virtual environment activated:

Reviewer: Does procedure contain a sequence of steps (then use an ordered list)? Or are they all alternatives to each other?

```suggestion
To activate the virtual environment, use `poetry shell`. This starts a new shell session with the virtual environment activated:
```

Author: @grischperl Can you help with this question?

Reviewer 2: The commands should be run in order. So, first run `poetry install` then run `poetry shell`.

Reviewer 2: No, not really. If you want to, you could formulate in in a way that there are steps:

1. `poetry shell` to activate the environment
2. `poetry run` to execute a command
3. `deactivate` to exit the environment

Reviewer 2: But in my opinion, it's also fine, as it's called "Activate the Virtual Environment" to say the relevant step is `poetry shell`.

And `poetry run` and `deactivate` is additional information

Author: Oh, now I get it, thank you :)

```bash
poetry shell
```

> You can now run Python and any installed packages in this shell. They will use the virtual environment. 

Reviewer: ```suggestion
Now, you can run Python and any installed packages in this shell. They will use the virtual environment. 
```

To exit the virtual environment, simply use the `deactivate` command.

Poetry automatically uses the virtual environment when you run, for example, the `poetry run` command.

Remember to run these commands in the root directory of your project, where the `pyproject.toml` file is located.

If you are a PyCharm user and want to use the virtual environment created by Poetry, follow the [configuration guides](https://www.jetbrains.com/help/pycharm/poetry.html).

## Use Poe the Poet as Task Runner

[Poe the Poet](https://poethepoet.natn.io/index.html) is a task runner that simplifies running common tasks in a Python project.

> To have the command available as `poetry poe <command>` as seen in the following examples, you need to install poe as a plugin to Poetry:

Reviewer: ```suggestion
To have the command available as `poetry poe <command>` (as seen in the following examples), install poe as a plugin to Poetry:
```

```bash
poetry self add 'poethepoet[poetry_plugin]'
```

> If this plugin is not installed, you may have to run poe as a script within the Poetry environment using `poetry run poe <command>`.

Reviewer: "may" - is this uncertain? Is there a chance that it works without the plugin, and even without running it as a script?

Author: @grischperl Can you help with this question?

Reviewer 2: "may" is wrong. If you don't have the plugin installed you have to use `poetry run poe <command>`. 

You can only use `poetry poe <command>` with the plugin installed.

## Development

### Redis Server 

> Before running the application, the Redis server must be provided. It is used to store the conversation with a large language model (LLM).

Reviewer: ```suggestion
Before running the application, you must provide the Redis server. It stores the conversation with a large language model (LLM).
```

> Therefore, **REDIS_URL** must be provided as an environment variable.

Reviewer: ```suggestion
Therefore, provide **REDIS_URL** as an environment variable.
```

> Here is the [documentation](https://github.tools.sap/kyma/ai-force/blob/main/docs/infrastructure/setup.md#15-redis) <!--the link must be replaced when the OS documentation is available --> on how to create a Redis server. 

Reviewer: ```suggestion
For details how to create a Redis server, read [Create Redis](https://github.tools.sap/kyma/ai-force/blob/main/docs/infrastructure/setup.md#15-redis) <!--replace the link when the OS documentation is available -->. 
```

In the /infrastructure/setup.md file, I suggest using more meaningful titles, like in this case, "Create Redis" (to match the two sub-procedures of this section).

For example, `REDIS_URL="redis://{host or ip}:6379"`

### Running Kyma Companion Locally

> You can start the FastAPI application with the following options:

Reviewer 2: maybe:
```suggestion
You can execute the Kyma Companion locally using the FastAPI framework with the following command:
```

```bash
poetry run fastapi dev src/main.py --port 8000
```

Or, with a poe task:

```bash
$ poetry poe run-local
```

> It is recommended to run Kyma Companion with Poetry as it activates and uses its virtual environment if not activated yet.

Reviewer: ```suggestion
It is recommended to run Kyma Companion with Poetry because it activates and uses its virtual environment if not activated yet.
```

> [!NOTE] You cannot run it with Python directly.

### Debugging

> As the companion uses the FastAPI framework, refer to the following documentation on how to debug the application with the respective IDE:

Reviewer: ```suggestion
Because the companion uses the FastAPI framework, read the following documentation on how to debug the application with the respective IDE:
```

* [PyCharm](https://www.jetbrains.com/help/pycharm/fastapi-project.html#create-project)
* [VS Code](https://code.visualstudio.com/docs/python/tutorial-fastapi)

### Configuration

> For local development, LLM models can be configured inside the `config/models.json` file.  

Reviewer: What is a LLM model? A large language model model? Is this something more abstract than a large language model?

Author: @grischperl Can you help with this question as well?

Reviewer 2: It's a [RAS Syndrome](https://en.wikipedia.org/wiki/RAS_syndrome)! Just "LLMs" or "large language models" would be correct, or even just "models"

> > [!NOTE] Don't use the <code>config/models.json</code> file to configure models in dev, stage, or prod clusters.

Reviewer: Why? 
If it's dangerous to do that, consider using a "Caution" and explain why it's a bad idea.

Author: @grischperl Can you help with this question as well?

Reviewer 2: The setup for dev, stage and prod will be different. Maybe this can be changed to simply say: "This file is for local development only." without mentioning dev, stage and prod?

## Create and Use Virtual Environments with Poetry

To create a virtual environment for the project, navigate to the project's root directory and run the following command:

```bash
poetry install
```

This creates a new virtual environment and installs the project's dependencies.

### Activate the Virtual Environment

To use the virtual environment, follow these steps:
1. Use `poetry shell` to start a new shell session with the virtual environment activated:

        ```bash
        poetry shell
        ```

   Now, you can run Python and any installed packages in this shell. They will use the virtual environment. 
2. Use `poetry run` to execute a command.
3. Use the `deactivate` command to exit the virtual environment.

> > [!NOTE] Run these commands in the root directory of your project, where the `pyproject.toml` file is located.

Reviewer: maybe this information is more helpful _before_ listing the commands.

If you are a PyCharm user and want to use the virtual environment created by Poetry, follow the [configuration guides](https://www.jetbrains.com/help/pycharm/poetry.html).

## Use Poe the Poet as Task Runner

[Poe the Poet](https://poethepoet.natn.io/index.html) is a task runner that simplifies running common tasks in a Python project.

To have the command available as `poetry poe <command>` (as seen in the following examples), install poe as a plugin to Poetry:

```bash
poetry self add 'poethepoet[poetry_plugin]'
```

If you don't install this plugin, you must run poe as a script within the Poetry environment using `poetry run poe <command>`.

## Development

### Redis Server 

Before running the application, you must provide the Redis server. It stores the conversation with a large language model (LLM).
Therefore, provide **REDIS_URL** as an environment variable.

> For details on how to create a Redis server, read [Create Redis](https://github.tools.sap/kyma/ai-force/blob/main/docs/infrastructure/setup.md#15-redis) <!--the link must be replaced when the OS documentation is available --> on how to create a Redis server. 

Reviewer: ```suggestion
For details on how to create a Redis server, read [Create Redis](https://github.tools.sap/kyma/ai-force/blob/main/docs/infrastructure/setup.md#15-redis). <!--the link must be replaced when the OS documentation is available -->
```

poetry poe lint
```

> Alternatively, you can also do it with `ruff check` directly, 

Reviewer: ```suggestion
Alternatively, you can also do it with `ruff check` directly, where Ruff may have a different version in a different virtual environment.
```

> where Ruff may have a different version in a different virtual environment.

Reviewer: no need to split this sentence 
```suggestion
```

> Linting errors can be fixed with the following command, which by default applies only the safe fixes:

Reviewer: ```suggestion
Linting errors can be fixed with the following command, which, by default, applies only the safe fixes:
```

poetry run pytest tests
```

## Release Process

Release testing and release creation are two separate processes. 

> You can find the release testing documentation in the [Contributor README](./docs/contributor/README.md) file.

Reviewer: ```suggestion
For details about release testing, read the [Contributor README](./docs/contributor/README.md).
```
